src/test_l1_dsHdmiIn.c

```diff
@@ -1725,22 +1819,30 @@ void test_l1_dsHdmiIn_negative_dsSetEdidVersion(void) {
     gTestID = 38;
     UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
     // Step 1: Call dsSetEdidVersion() without initializing the HDMI input sub-system
+    #ifdef ENABLE_ENHANCED_ERROR_CODE
     UT_ASSERT_EQUAL(dsSetEdidVersion(dsHDMI_IN_PORT_0, HDMI_EDID_VER_14), dsERR_NOT_INITIALIZED);
+    #endif
 
     // Step 2: Initialize the HDMI input sub-system using dsHdmiInInit()
     UT_ASSERT_EQUAL(dsHdmiInInit(), dsERR_NONE);
 
     // Step 3: Call dsSetEdidVersion() with invalid inputs (dsHDMI_IN_PORT_MAX)
+    #ifdef ENABLE_ENHANCED_ERROR_CODE
     DS_ASSERT_AUTO_TERM_NUMERICAL(dsSetEdidVersion(dsHDMI_IN_PORT_MAX, HDMI_EDID_VER_14), dsERR_INVALID_PARAM);
+    #endif
 
     // Step 4: Call dsSetEdidVersion() with invalid inputs (HDMI_EDID_VER_MAX)
+    #ifdef ENABLE_ENHANCED_ERROR_CODE
     DS_ASSERT_AUTO_TERM_NUMERICAL(dsSetEdidVersion(dsHDMI_IN_PORT_0, HDMI_EDID_VER_MAX), dsERR_INVALID_PARAM);
```

Comment:
In the else case, we need to handle the expected dsERR_GENERAL.

Reply:
done

Reply:
done